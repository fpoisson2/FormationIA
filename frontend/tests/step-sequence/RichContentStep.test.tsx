import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import {
  STEP_COMPONENT_REGISTRY,
  StepSequenceContext,
  type StepComponentProps,
} from "../../src/modules/step-sequence";
import {
  RichContentStep,
  type RichContentStepConfig,
} from "../../src/modules/step-sequence/modules";

function renderRichContentStep(
  isEditMode: boolean,
  config: RichContentStepConfig
) {
  const onAdvance = vi.fn();
  const onUpdateConfig = vi.fn();

  const props: StepComponentProps = {
    definition: { id: "rich", component: "rich-content", config },
    config,
    payload: undefined,
    isActive: true,
    isEditMode,
    onAdvance,
    onUpdateConfig,
  };

  const view = render(
    <StepSequenceContext.Provider
      value={{
        stepIndex: 0,
        isEditMode,
        onAdvance,
        onUpdateConfig,
      }}
    >
      <RichContentStep {...props} />
    </StepSequenceContext.Provider>
  );

  return { container: view.container, onUpdateConfig };
}

describe("RichContentStep", () => {
  it("registers itself under the rich-content key", () => {
    expect(STEP_COMPONENT_REGISTRY["rich-content"]).toBe(RichContentStep);
  });

  it("matches the read-only snapshot", () => {
    const { container } = renderRichContentStep(false, {
      title: "Introduction",
      body: "Bienvenue sur la plateforme !\nProfitez de votre parcours.",
      media: [
        {
          id: "m1",
          url: "https://cdn.example.com/hero.png",
          alt: "Capture d'écran",
          caption: "Interface principale",
        },
      ],
      sidebar: {
        type: "tips",
        title: "Astuces",
        tips: ["Prenez des notes", "Planifiez vos séances"],
      },
    });

    expect(container).toMatchSnapshot();
  });

  it("matches the editable snapshot", () => {
    const { container } = renderRichContentStep(true, {
      title: "Atelier",
      body: "Complétez les informations demandées ci-dessous.",
      media: [
        {
          id: "m1",
          url: "https://cdn.example.com/form.png",
          alt: "Formulaire",
        },
      ],
      sidebar: {
        type: "checklist",
        title: "Checklist",
        items: [
          { id: "i1", label: "Ajouter un média", checked: true },
          { id: "i2", label: "Renseigner la description" },
        ],
      },
      onChange: vi.fn(),
    });

    expect(container).toMatchSnapshot();
  });
});
