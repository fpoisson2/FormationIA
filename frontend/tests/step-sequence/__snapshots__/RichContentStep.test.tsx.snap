// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RichContentStep > matches the editable snapshot 1`] = `
<div>
  <div
    class="space-y-8"
  >
    <div
      class="grid gap-6 md:grid-cols-[2fr,1fr]"
    >
      <section
        class="space-y-4"
      >
        <label
          class="block space-y-2"
        >
          <span
            class="text-sm font-medium text-slate-700"
          >
            Titre
          </span>
          <input
            class="w-full rounded-md border border-slate-300 p-2"
            placeholder="Titre de l'étape"
            value="Atelier"
          />
        </label>
        <label
          class="block space-y-2"
        >
          <span
            class="text-sm font-medium text-slate-700"
          >
            Contenu
          </span>
          <textarea
            class="h-40 w-full rounded-md border border-slate-300 p-2"
            placeholder="Description détaillée"
          >
            Complétez les informations demandées ci-dessous.
          </textarea>
        </label>
        <div
          class="space-y-2"
        >
          <p
            class="text-sm font-medium text-slate-700"
          >
            Médias
          </p>
          <div
            class="space-y-4"
          >
            <div
              class="rounded-md border border-slate-200 p-4 shadow-sm"
            >
              <div
                class="grid gap-3 md:grid-cols-2"
              >
                <label
                  class="block space-y-1"
                >
                  <span
                    class="text-xs font-medium text-slate-600"
                  >
                    URL de l'image
                  </span>
                  <input
                    class="w-full rounded-md border border-slate-300 p-2"
                    placeholder="https://..."
                    value="https://cdn.example.com/form.png"
                  />
                </label>
                <label
                  class="block space-y-1"
                >
                  <span
                    class="text-xs font-medium text-slate-600"
                  >
                    Texte alternatif
                  </span>
                  <input
                    class="w-full rounded-md border border-slate-300 p-2"
                    placeholder="Description"
                    value="Formulaire"
                  />
                </label>
              </div>
              <label
                class="mt-3 block space-y-1"
              >
                <span
                  class="text-xs font-medium text-slate-600"
                >
                  Légende (optionnelle)
                </span>
                <input
                  class="w-full rounded-md border border-slate-300 p-2"
                  placeholder="Ajouter une légende"
                  value=""
                />
              </label>
              <button
                class="mt-3 text-sm text-red-600"
                type="button"
              >
                Supprimer ce média
              </button>
            </div>
            <div
              class="flex flex-col gap-2 sm:flex-row sm:items-center"
            >
              <button
                class="rounded-md border border-slate-300 px-4 py-2 text-sm"
                type="button"
              >
                Ajouter un média par URL
              </button>
              <label
                class="text-sm text-slate-600"
              >
                <span
                  class="mr-2"
                >
                  ou importer un fichier
                </span>
                <input
                  accept="image/*"
                  type="file"
                />
              </label>
            </div>
          </div>
        </div>
      </section>
      <section
        class="space-y-4"
      >
        <label
          class="block space-y-2"
        >
          <span
            class="text-sm font-medium text-slate-700"
          >
            Type de sidebar
          </span>
          <select
            class="w-full rounded-md border border-slate-300 p-2"
          >
            <option
              value="none"
            >
              Aucune
            </option>
            <option
              value="tips"
            >
              Astuces
            </option>
            <option
              value="checklist"
            >
              Checklist
            </option>
          </select>
        </label>
        <div
          class="space-y-4 rounded-md border border-slate-200 p-4 shadow-sm"
        >
          <label
            class="block space-y-1"
          >
            <span
              class="text-xs font-medium text-slate-600"
            >
              Titre de la sidebar
            </span>
            <input
              class="w-full rounded-md border border-slate-300 p-2"
              placeholder="Titre optionnel"
              value="Checklist"
            />
          </label>
          <div
            class="space-y-3"
          >
            <div
              class="rounded-md border border-slate-200 p-3"
            >
              <label
                class="mb-2 flex items-center gap-2 text-xs font-medium text-slate-600"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                <span>
                  Élément validé
                </span>
              </label>
              <input
                class="w-full rounded-md border border-slate-300 p-2 text-sm"
                placeholder="Tâche 1"
                value="Ajouter un média"
              />
              <button
                class="mt-2 text-xs text-red-600"
                type="button"
              >
                Supprimer
              </button>
            </div>
            <div
              class="rounded-md border border-slate-200 p-3"
            >
              <label
                class="mb-2 flex items-center gap-2 text-xs font-medium text-slate-600"
              >
                <input
                  type="checkbox"
                />
                <span>
                  Élément validé
                </span>
              </label>
              <input
                class="w-full rounded-md border border-slate-300 p-2 text-sm"
                placeholder="Tâche 2"
                value="Renseigner la description"
              />
              <button
                class="mt-2 text-xs text-red-600"
                type="button"
              >
                Supprimer
              </button>
            </div>
          </div>
          <button
            class="text-sm text-blue-600"
            type="button"
          >
            Ajouter un élément
          </button>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`RichContentStep > matches the read-only snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-6 md:flex-row"
  >
    <article
      class="flex-1 space-y-6"
    >
      <h2
        class="text-2xl font-semibold text-slate-900"
      >
        Introduction
      </h2>
      <p
        class="whitespace-pre-line text-base text-slate-700"
      >
        Bienvenue sur la plateforme !
Profitez de votre parcours.
      </p>
      <div
        class="grid gap-4 md:grid-cols-2"
      >
        <figure
          class="rounded-lg border border-slate-200 bg-white p-4 shadow-sm"
        >
          <img
            alt="Capture d'écran"
            class="w-full rounded-md object-cover"
            src="https://cdn.example.com/hero.png"
          />
          <figcaption
            class="mt-2 text-sm text-slate-600"
          >
            Interface principale
          </figcaption>
        </figure>
      </div>
    </article>
    <aside
      class="rounded-md border border-slate-200 bg-slate-50 p-4 text-sm"
    >
      <h3
        class="mb-2 font-semibold text-slate-700"
      >
        Astuces
      </h3>
      <ul
        class="list-disc space-y-1 pl-4 text-slate-600"
      >
        <li>
          Prenez des notes
        </li>
        <li>
          Planifiez vos séances
        </li>
      </ul>
    </aside>
  </div>
</div>
`;
